From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: HaHaWTH <102713261+HaHaWTH@users.noreply.github.com>
Date: Sat, 26 Oct 2024 14:04:54 +0800
Subject: [PATCH] Leaf: Remove stream in GolemSensor

Stream operations in GolemSensor is really expensive and takes
up 80% time per method call.
Before: 192ms
After: 17ms

diff --git a/net/minecraft/world/entity/ai/sensing/GolemSensor.java b/net/minecraft/world/entity/ai/sensing/GolemSensor.java
index 84d9e2a43adbabda8401e8ad8dd8d87f7dbeeea7..6d85ff6cf8902e0e5ecf5966de58e7b7aaf625d0 100644
--- a/net/minecraft/world/entity/ai/sensing/GolemSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/GolemSensor.java
@@ -33,8 +33,16 @@ public class GolemSensor extends Sensor<LivingEntity> {
 
     public static void checkForNearbyGolem(LivingEntity livingEntity) {
         Optional<List<LivingEntity>> memory = livingEntity.getBrain().getMemory(MemoryModuleType.NEAREST_LIVING_ENTITIES);
-        if (!memory.isEmpty()) {
-            boolean flag = memory.get().stream().anyMatch(entity -> entity.getType().equals(EntityType.IRON_GOLEM));
+        if (memory.isPresent()) {
+            // Leaf start - Remove stream in GolemSensor
+            boolean flag = false;
+            for (LivingEntity entity : memory.get()) {
+                if (entity.getType().equals(EntityType.IRON_GOLEM)) {
+                    flag = true;
+                    break;
+                }
+            }
+            // Leaf end - Remove stream in GolemSensor
             if (flag) {
                 golemDetected(livingEntity);
             }
